dbms/src/Interpreters/loadMetadata.cpp

```diff
@@ -166,7 +175,22 @@ void loadMetadata(Context & context)
             load_database(context, db_name, meta_file, &load_tables_thread_pool, has_force_restore_data_flag);
         };
 
-        load_databases_thread_pool.scheduleOrThrowOnError(task);
+        try {
+            load_databases_thread_pool.scheduleOrThrowOnError(task);
+        } catch (const Exception & e) {
+            if (e.code() == ErrorCodes::CANNOT_SCHEDULE_TASK)
+            {
+                // If we cannot schedule task, we will run it in current thread.
+                // This is to avoid the case that we cannot load any database.
+                LOG_ERROR(log, "scheduleOrThrowOnError failed and we try to run it in current thread, error code = {}, e.displayText() = {}", e.code(), e.displayText());
```

Comment:
Maybe just throw is fine, because:

1. It is our fault if we use toooo many threads when there are too many tables. Instead we must control the max thread we want to use.
2. It is users' fault if max thread is set to a value too low.